### DISCUSSION TITLE
sistema de cambio de escenas para tpi

### DISCUSSION THREAD

**Discussion by rinrot** [INGCOM-UNRN-P1/cursada-2023#224] [Preguntas y respuestas] [answered]
Hola! estoy haciendo una especie de novela/crea tu propia aventura y estoy todavia intentando cranear una funcion/sistema para poder pasar de dialogo/escena intentando evitar lo mas posible anidar el codigo como 30 veces con los caminos. se me habia ocurrido hacer una estructura para eso, pero siempre hay una cantidad de opciones diferentes para cada texto. ᵃʸᵘᵈᵃ

**Comment by martinvilu** [comment, marked as answer]
A mi parecer, la clave es apoyarte en los archivos, que los mismos contengan además del texto, las conexiones con otros para armar las opciones, esto da una flexibilidad enorme y hace que la historia sea completamente independiente del “motor” de juego.

En lugar de que las opciones sean principalmente código, que el mismo archivo te lleve al siguiente.
![imagen](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/56625/379d011b-85fb-4f91-82af-a7f55c05fd19)

Un archivo puede ser algo como

```
>>T
¡Prepárate para una emocionante odisea en el mundo de la programación en C! Eres un intrépido programador que, por alguna razón desconocida, ha sido transportado misteriosamente a un reino digital lleno de desafíos y peligros. Tu única manera de regresar a casa es navegar a través de este mundo programático y superar obstáculos codificados.

A medida que avanzas, te enfrentarás a monstruos de errores sintácticos, resolverás acertijos lógicos para desbloquear puertas de condicionales y te adentrarás en bosques oscuros de loops infinitos. Cada decisión que tomes afectará el curso de tu aventura y la eficiencia de tu código. ¿Optarás por el camino directo usando funciones predefinidas o te aventurarás a escribir tu propia solución desde cero?

En este juego tu destino está en tus manos. ¿Estás listo para la aventura de programación en C de tu vida? ¡Que comience la codificación!
<<T
>>O
 >M Por donde queres empezar
 >R "La Ruta de las Estructuras de Control" [camino_1.txt]
 >R "La senda del Algoritmo" [camino_2.txt]
<<O
```

Use `>>` para indicar y marcar las diferentes partes del archivo (como una combinacion de caracteres poco probables de ver en los textos); primero el texto que se le muestra al jugador
y después, las opciones, pero estas incluyen el archivo al que se “saltara” de elegir esa opción. (casi como una `struct` ;-) )

# Recuerden que son solo sugerencias, no hay ninguna obligación de hacerlo de esta manera

**Reply by rinrot** [reply]
yo estaba haciendo algo parecido imprimiendo las preguntas dentro del txt, pero todavia no se ocurre la manera en la que poner el nombre del siguiente archivo en el txt lleve a su uso. cuando tenga algo mas concreto lo subo aca. Gracias por la idea profe!
